website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,371 @@
+- `indexing` - [HoodieIndexCommitMetadata](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieIndexCommitMetadata.avsc)
+
+Reconciling all the actions in the timeline, the state of the Hudi dataset can be re-created at any instant of time.    
+
+​	
+
+## **Metadata**
+
+Hudi automatically extracts the physical data statistics and stores the metadata along with the data to improve write and query performance. Hudi Metadata is an internally-managed table which organizes the table metadata under the base path *.hoodie/metadata.* The data format used is similar to the merge-on-read data format. Every record stored in the metadata table is a Hudi row and hence has partitioning key and row key specified. Following are the metadata table partitions
+
+- **files** - Partition path to file name index. Key for the Hudi record is the partition path and the actual record is a map of file name to an instance of [HoodieMetadataFileInfo](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L34) . The files index can be used to do file listing and do filter based pruning of the scanset during query
+
+- **bloom_filters** - Bloom filter index to help map a record key to the actual file. The Hudi key is `hash(partition name) + hash(file name)` and the actual payload is an instance of [HudiMetadataBloomFilter](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L66) . Bloom filter helps with identifying the files to update and optimize joins with row key and point queries with row key filter.
+- **column_stats** - contains statistics of all the columns for all the rows in the table. This enables file grained file pruning for filters and join conditions in the query. The actual payload is an instance of [HoodieMetadataColumnStats](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L101). 
+
+
+
+## Data Organization
+
+As mentioned in the data model, data is partitioned coarsely through a directory hierarchy based on the partition path configured. Within each partition the data is physically stored as **base and log files** and organized into logical concepts as **File groups and File slices**. The logical concepts will be referred by the writer / reader requirements. 
+
+**File group** - Groups multiple versions of a base file. File group is uniquely identified by a File id. Each version corresponds to the commit instants timestamp recording updates to rows in the file. The base file are stored in open source data formats like  Apache Parquet, Apache ORC, Apache HBase HFile etc.
+
+**File slice** - A File group can further be split into multiple files. A base file corresponding to commit timestamp and a set of log files that batches the deletes/updates to the base file and each log file corresponds to a delta commit timestamp and the delta commits happen after the base file commit in the Hudi timeline. 
```

Comment:
nit: "into multiple slices"

Reply:
> A base file corresponding to commit timestamp and a set of log files that batches the deletes/updates to the base file and each log file corresponds to a delta commit timestamp and the delta commits happen after the base file commit in the Hudi timeline.

 Definition seems a bit wordy, i'd suggest something along the lines of: "Each file slice w/in the file-group is uniquely identified by commit's timestamp that introduced it (<some examples here>). In case of COW, file-slice is simply just another version of the base-file. In case of MOR it's a combination of the base-file along with log-files attached to it"